From 900cc60d9162a4857fd9fa00761d9641c1d175b2 Mon Sep 17 00:00:00 2001
From: Kuppuswamy Sathyanarayanan <sathyanarayanan.kuppuswamy@linux.intel.com>
Date: Mon, 21 May 2018 14:58:35 -0700
Subject: [PATCH 1023/1240] Revert "usb: pci-quirks: register USB mux found on
 Cherrytrail SOC"

This reverts commit da8c9c24a6602c0e54ee89172cabb1d13fe2c8ad.

Signed-off-by: Kuppuswamy Sathyanarayanan <sathyanarayanan.kuppuswamy@linux.intel.com>
---
 drivers/usb/host/pci-quirks.c | 25 +------------------------
 1 file changed, 1 insertion(+), 24 deletions(-)

diff --git a/drivers/usb/host/pci-quirks.c b/drivers/usb/host/pci-quirks.c
index 1d30f88..e1faee1 100644
--- a/drivers/usb/host/pci-quirks.c
+++ b/drivers/usb/host/pci-quirks.c
@@ -15,7 +15,6 @@
 #include <linux/export.h>
 #include <linux/acpi.h>
 #include <linux/dmi.h>
-#include <linux/extcon/intel_usb_mux.h>
 #include "pci-quirks.h"
 #include "xhci-ext-caps.h"
 
@@ -1201,31 +1200,9 @@ static void quirk_usb_handoff_xhci(struct pci_dev *pdev)
 	writel(val, base + ext_cap_offset + XHCI_LEGACY_CONTROL_OFFSET);
 
 hc_init:
-	if (pdev->vendor == PCI_VENDOR_ID_INTEL) {
+	if (pdev->vendor == PCI_VENDOR_ID_INTEL)
 		usb_enable_intel_xhci_ports(pdev);
 
-		/*
-		 * Initialize the internal mux that shares a port between USB
-		 * Device Controller and xHCI on platforms that have it.
-		 */
-#define XHCI_INTEL_VENDOR_CAPS 192
-		ext_cap_offset = xhci_find_next_ext_cap(base,
-							XHCI_HCC_PARAMS_OFFSET,
-							XHCI_INTEL_VENDOR_CAPS);
-		if (ext_cap_offset) {
-			struct intel_usb_mux *mux;
-			struct resource r;
-
-			r.start = pci_resource_start(pdev, 0) + 0x80d8;
-			r.end   = r.start + 8;
-			r.flags = IORESOURCE_MEM;
-
-			mux = intel_usb_mux_register(&pdev->dev, &r);
-			if (IS_ERR(mux))
-				dev_err(&pdev->dev, "failed to register mux\n");
-		}
-	}
-
 	op_reg_base = base + XHCI_HC_LENGTH(readl(base));
 
 	/* Wait for the host controller to be ready before writing any
-- 
2.7.4

